src/crosschain/swapExactIn/stonfiSwap.ts

```diff
@@ -0,0 +1,73 @@
+import { SwapExactInParams, SwapExactInResult } from '../types'
+import { StonfiTrade } from '../trade'
+import { getTonTokenAddress, isTonChainId } from '../chainUtils'
+import { StonApiClient } from '@ston-fi/api'
+
+export async function isStonfiSwapSupported(context: SwapExactInParams): Promise<boolean> {
+    const stonClient = new StonApiClient()
+    const { tokenAmountIn, tokenOut } = context
+
+    if (!isTonChainId(tokenAmountIn.token.chainId) || !isTonChainId(tokenOut.chainId)) {
+        return false
+    }
+
+    try {
+        await Promise.all([
```

Comment:
it is bad idea to make isStonfiSwapSupported async.

it should be fast.

lets move this check to  `stonfiSwap` func